File: summary/ruby/7_greatest_common_divisor.rb
def gcd(a, b)
  while a >= 1 && b >= 1
    if a >= b
      a = a % b
    else
      b = b % a
    end
  end

  a >= 1 ? a : b
end

a, b = gets.chomp.split.map(&:to_i)
puts "#{gcd(a, b)}"

File: summary/dart/7_greatest_common_divisor.dart
int gcd(int a, int b) {
  while (b != 0) {
    int temp = b;
    b = a % b;
    a = temp;
  }
  return a;
}

void main() {
  print(gcd(48, 18)); // 6
  print(gcd(100, 25)); // 25
  print(gcd(7, 3)); // 1
}
